File: uidesigner/com.nokia.carbide.cpp.uiq.ui/src/com/nokia/carbide/cpp/uiq/ui/vieweditor/NamedGroup.java
/*
* Copyright (c) 2009 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of the License "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description: 
*
*/
/* START_USECASES: CU6 END_USECASES */
package com.nokia.carbide.cpp.uiq.ui.vieweditor;
/**
 * Class the represents the Named group instance in the local model.
 *
 */
public class NamedGroup extends CommandElementBase{
	private int children = 0;
	/**
	 * Class constructor
	 * @param name
	 */
	public NamedGroup(String name){
		this.setName(name);
	}
	/**
	 * Adds a child to the children counter
	 */
	public void addChildren(){
		children++;
	}
	/**
	 * Removes a child to the children counter
	 */
	public void removeChildren(){
		children--;
	}
	/**
	 * Returns the number of children
	 * @return
	 */
	public int getChildren(){
		return children;
	}
}

File: uidesigner/com.nokia.carbide.cpp.uiq.components/components/command_related/NamedGroup/NamedGroup.js
/*
* Copyright (c) 2007 Nokia Corporation and/or its subsidiary(-ies).
* All rights reserved.
* This component and the accompanying materials are made available
* under the terms of the License "Eclipse Public License v1.0"
* which accompanies this distribution, and is available
* at the URL "http://www.eclipse.org/legal/epl-v10.html".
*
* Initial Contributors:
* Nokia Corporation - initial contribution.
*
* Contributors:
*
* Description: 
* START_USECASES: CU11 END_USECASES
*
*
*/


include("../../implLibrary.js")  

function NamedGroup() {
}

////////////////////////////////////////////////////////////////////////////////
//IComponentValidator

NamedGroup.prototype.validate = function(instance) {
	properties = instance.properties;
	if (instance != null || instance != "") {
		if(properties.text.length > 255  ){
			return [createSimpleModelError(instance, 
					"text",
					formatString(lookupString("validate.text"),
					[instance.name ]),
					null )
				   ];
		}				
		if(properties.shortText.length > 255  ){
			return [createSimpleModelError(instance, 
					"shortText",
					formatString(lookupString("validate.shortText"),
					[instance.name ]),
					null )
				   ];
		}				
	}
	return null;
}

NamedGroup.prototype.queryPropertyChange = function(instance, propertyPath, newValue) {
	return null;
}
